instrumentation/spring/spring-web/spring-web-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springweb/v3_1/WebApplicationContextInstrumentation.java

```diff
@@ -84,7 +84,8 @@ public static void onEnter(@Advice.Argument(0) ConfigurableListableBeanFactory b
           ((BeanDefinitionRegistry) beanFactory)
               .registerBeanDefinition("otelAutoDispatcherFilter", beanDefinition);
         } catch (ClassNotFoundException ignored) {
-          // Ignore
+          ignored.printStackTrace(System.out);
+          throw new IllegalStateException(ignored); // TODO: for debugging
```

Comment:
@JonasKunz this is where it currently fails as the proxy class is not found, even if we load it with the "correct" classloader which is the application one, so it should be able to load the proxy class. Adding the exception allows to un-hide the silent failure.